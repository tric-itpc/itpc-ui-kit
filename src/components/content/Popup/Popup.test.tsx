import React from "react"

import { render, screen } from "@testing-library/react"

import { Popup } from "./index"

describe("Компонент Popup", () => {
  test("Рендер с обязательными пропсами", () => {
    render(<Popup title="Заголовок" isOpen />)

    expect(screen.getByText("Заголовок")).toBeInTheDocument()
    expect(screen.getByRole("popup")).toHaveClass("itpc-popup_opened")
  })

  test("Закрыт при isOpen=false", () => {
    render(<Popup isOpen={false} title="Заголовок" />)

    expect(screen.getByRole("popup", { hidden: true })).toHaveClass(
      "itpc-popup_closed"
    )
  })

  test("Рендер с иконкой закрытия", () => {
    render(
      <Popup
        iconClose={<span data-testid="close-icon">×</span>}
        title="Заголовок"
        isOpen
      />
    )

    expect(screen.getByTestId("close-icon")).toBeInTheDocument()
  })

  test("Рендер с контентом", () => {
    render(
      <Popup title="Заголовок" isOpen>
        <p>Контент попапа</p>
      </Popup>
    )

    expect(screen.getByText("Контент попапа")).toBeInTheDocument()
  })

  test("Применение дополнительных классов", () => {
    render(<Popup className="custom-class" title="Заголовок" isOpen />)

    expect(screen.getByRole("popup")).toHaveClass("custom-class")
  })

  test("Применение позиции", () => {
    render(<Popup position="top-left" title="Заголовок" isOpen />)

    expect(screen.getByRole("popup")).toHaveStyle({ left: "1rem", top: "1rem" })
  })

  test("Применение размера normal", () => {
    render(
      <Popup size="normal" title="Заголовок" isOpen>
        <p>Контент попапа</p>
      </Popup>
    )

    expect(screen.getByText("Контент попапа")).toBeInTheDocument()
  })

  test("Применение варианта success", () => {
    render(<Popup title="Заголовок" variant="success" isOpen />)

    expect(screen.getByRole("popup")).toHaveClass("itpc-popup_color_success")
  })

  test("Рендер с иконкой закрытия (снапшот)", () => {
    const { asFragment } = render(
      <Popup
        iconClose={<span data-testid="close-icon">×</span>}
        title="Заголовок"
        isOpen
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Рендер с контентом (снапшот)", () => {
    const { asFragment } = render(
      <Popup title="Заголовок" isOpen>
        <p>Контент попапа</p>
      </Popup>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Применение дополнительных классов (снапшот)", () => {
    const { asFragment } = render(
      <Popup className="custom-class" title="Заголовок" isOpen />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Применение позиции (снапшот)", () => {
    const { asFragment } = render(
      <Popup position="top-left" title="Заголовок" isOpen />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Применение размера normal (снапшот)", () => {
    const { asFragment } = render(
      <Popup size="normal" title="Заголовок" isOpen>
        <p>Контент попапа</p>
      </Popup>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("Применение варианта success (снапшот)", () => {
    const { asFragment } = render(
      <Popup title="Заголовок" variant="success" isOpen />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
