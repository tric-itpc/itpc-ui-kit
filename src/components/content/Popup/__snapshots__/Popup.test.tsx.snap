// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Popup Применение варианта success (снапшот) 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened itpc-popup_color_success"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Применение варианта success 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened itpc-popup_color_success"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Применение дополнительных классов (снапшот) 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened custom-class"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Применение дополнительных классов 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened custom-class"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Применение позиции (снапшот) 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: 1rem; top: 1rem;"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Применение позиции 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: 1rem; top: 1rem;"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Применение размера normal (снапшот) 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
    <p>
      Контент попапа
    </p>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Применение размера normal 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
    <p>
      Контент попапа
    </p>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Рендер с иконкой закрытия (снапшот) 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
      <span
        data-testid="close-icon"
      >
        ×
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Рендер с иконкой закрытия 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
      <span
        data-testid="close-icon"
      >
        ×
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Рендер с контентом (снапшот) 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
    <p>
      Контент попапа
    </p>
  </div>
</DocumentFragment>
`;

exports[`Компонент Popup Рендер с контентом 1`] = `
<DocumentFragment>
  <div
    class="itpc-popup itpc-popup_opened"
    role="popup"
    style="left: calc(50% - 0px); top: calc(50% - 0px);"
  >
    <div
      class="itpc-popup__header"
    >
      <p
        class="itpc-popup__title"
      >
        Заголовок
      </p>
    </div>
    <p>
      Контент попапа
    </p>
  </div>
</DocumentFragment>
`;
